### ISSUE TITLE
[REQUEST] Matter (Project CHIP) compatible "OpenThread Border Router" (OTBR) firmware for Elelabs adapters?

### ISSUE THREAD

**Issue by Hedda** [Elelabs/elelabs-zigbee-ezsp-utility#15]
This is feature request is for Matter (Project CHIP) compatible "Thread Border Router" OpenThread firmware for Elelabs adapters.

@NilsBohr Great to hear that you will release OpenThread firmware for your existing and future Elelabs adapters based on Silabs EFR32! You might by the way consider clarifying that on your site blog/news page and spread the release to tech sites and bloggers/vloggers.

To be honest I have not really read up on how Matter works with a Thread Border Router so maybe your firmware will be compatible?

There is also not really any Thread based Matter devices available so not a great demand though it would not hurt Elelabs sales so get in early and start providing OpenThread firmware for Elelabs adapters that enable "OpenThread Border Router" (OTBR) for Matter.

Quick summary of Matter over Thread as I understand it is that it will work very similar to a Zigbee network (with the application and devices even using the Zigbee Cluster Service for device classes, attributes and commands) but instead of flashing a Silabs EFR32 based adapter (like the Elelabs Zigbee USB dongle) with a Zigbee stack firmware configured as a "Zigbee Coordinator" which will be the application's interface to control all Zigbee devices on a Zigbee network, users of Matter over Thread will need to buy and flash a compatible adapter with an OpenThread firmware and configure that as an "OpenThread Border Router" ("OTBR" or just "Thread Border Router"") in order to configure a Thread network that the Matter application connectivity IP-based protocol can communicate over in order to control Matter devices that are based on Thread, (e.g. practically any battery-operated Matter device like sensors such as a temperature sensor, motion sensor, or window and door sensor).

References:

https://buildwithmatter.com/

https://www.threadgroup.org/news-events/blog/ID/287/Thread-Matter-And-CHIP--This-Glossary-Will-Quickly-Bring-You-Up-To-Speed

https://github.com/project-chip/connectedhomeip

https://github.com/project-chip/connectedhomeip/tree/master/third_party/openthread

https://github.com/project-chip/connectedhomeip/tree/master/third_party

https://github.com/project-chip/connectedhomeip/tree/master/third_party/efr32_sdk

https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/openthread_border_router_pi.md

https://openthread.io/guides/border-router

https://openthread.io/guides/border-router/build

https://openthread.io/codelabs/openthread-border-router

http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-3244/matter/openthread_border_router_pi.html

http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-3244/matter/openthread_rcp_nrf_dongle.html

https://www.silabs.com/documents/public/application-notes/an1256-using-sl-rcp-with-openthread-border-router.pdf

**Comment by NilsBohr** [comment]
@Hedda 
The Thread firmware, that we currently OpenThread Border router compatible.
It also works with OpenThread Pyspinel CMD line tool.

So, no additional changes there.
However, we will most probably release a pre-built RPi Image with OpenThread Border router to immediately check it out.

**Comment by NilsBohr** [comment]
@Hedda 
```
You might by the way consider clarifying that on your site blog/news page and spread the release to tech sites and bloggers/vloggers.
```
Thank you for the suggestion. As you mentioned, it's not really possible to test it, as not many devices out there yet.
But, ye, we will consider this.

**Comment by Hedda** [comment]
Yes I understand there are  ot any Thread based Matter devices available you can buy as of yet so currently developers need to really make their own. Nordic Semi look to have some instructions for doing so using inexpensive hardware:

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_matter_creating_accessory.html

https://github.com/project-chip/zap

**Comment by NilsBohr** [comment]
Thank you.
Yes, in our development lab we had been "playing" with Matter for a while now.
But it's still very developer oriented, and not yet easily available for DIY enthusiasts.

**Comment by Hedda** [comment]
Yeah, though with Matter (Project CHIP) now having over 150 members it sounds as if a lot of Matter devices will arrive next year.

For when those available they might not become popular with DIY home automation software enthusiasts until a skilled developer community code equivalent of [Zigbee2MQTT](https://www.zigbee2mqtt.io/) or the [ZHA integration for Home Assistant](https://www.home-assistant.io/integrations/zha) as a Matter border router integration.

Interestingly the founders of Home Assistant working at Nabu Casa have said that they plan on supporting Matter using the onboard ERF32 based radio that they will ship in their Amber hardware so Elelabs hardware will likely be compatible with any upcoming implementation:

https://www.home-assistant.io/blog/2021/09/13/home-assistant-amber/

https://www.crowdsupply.com/nabu-casa/home-assistant-amber
